### PR TITLE
Allow OS to define config highlighting (pfSense)

### PR DESCRIPTION
**Pull request by brownowski**
Fix display for pfsense config files.
 - use xml language formatting for pfsense os
 - don't use htmlspecialchars_decode as xml tags won't show

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
htmlspecialchars_decode() doesn't remove tags it converts things like &quot; to "

#8847

**Comment by brownowski** [comment]
Yes, but by decoding the codes which are returned by the function, the xml tags in the config file are returned to the browser as tags instead of special codes and therefore don't display properly.

**Comment by brownowski** [comment]
You can probably disregard this pull request.  I don't know what, but something seems to have changed from when I was testing originally and the original code now works ¯\_(ツ)_/¯

I've updated my patch for xml highlighting of the pfsense config file, which now seems to require me running htmlspecialchars_decode on $text before sending to GeSHi.  If I don't decode the special chars before GeSHi, I only get the start and close angle brackets highlighted due to them being special char encoded.  

However, the original code now seems to show the config file with the tags correctly, albeit with no syntax highlighting.

**Comment by brownowski** [comment]
There might still be something here.  If I do a diff now I'm missing the xml tags like I was before.

Pre-patch:  htmlspecialchars_decode on $geshi->parse_code()
pre-patch

Post-patch: htmlspecialchars_decode on $text pre-GeSHi 
post-patch

File: includes/html/pages/device/showconfig.inc.php [hunk 1]
```diff
@@ -318,16 +318,18 @@
     if (!empty($text)) {
         if (isset($previous_config)) {
             $language = 'diff';
+        } elseif ($device['os'] == 'pfsense') {
```
Comment (laf) [inline includes/html/pages/device/showconfig.inc.php]: Should we move this to the device definition? So have `config_language: xml` in the pfsense definition then fall back to ios for all others?

Comment (murrant) [reply includes/html/pages/device/showconfig.inc.php]: config_highlighting would be my naming

Comment (laf) [reply includes/html/pages/device/showconfig.inc.php]: Even better :)

Comment (laf) [reply includes/html/pages/device/showconfig.inc.php]: @brownowski Can you implement this please?

File: includes/html/pages/device/showconfig.inc.php [hunk 2]
```diff
@@ -318,16 +318,18 @@
     if (!empty($text)) {
         if (isset($previous_config)) {
             $language = 'diff';
+        } elseif (isset(Config::get('os')[$device['os']]['config_highlighting'])) {
```
Comment (murrant) [inline includes/html/pages/device/showconfig.inc.php]: Do this like this:
```php
$language = isset($previous_config) ? 'diff' : Config::getOsSetting($device['os'], 'config_highlighting', 'ios');
```

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-56-release-changelog-september-2019/9671/1

**Pull request merged**